app/views/associate_ucr.scala.html

```diff
@@ -14,29 +14,43 @@
  * limitations under the License.
  *@
 
-@import forms.AssociateDucr
+@import forms.AssociateUcr
 @import forms.MucrOptions
+@import views.components.fields.RadioOption
 @import views.Title
+@import forms.AssociateKind
 
 @this(main_template: views.html.main_template)
 
-@(form: Form[AssociateDucr], mucrOptions: MucrOptions)(implicit request: Request[_], messages: Messages)
+@(form: Form[AssociateUcr], mucrOptions: MucrOptions)(implicit request: Request[_], messages: Messages)
 
 
-@main_template(title = Title(messages("associateDucr.title", mucrOptions.mucr))) {
+@main_template(title = Title(messages("associate.ucr.title", mucrOptions.mucr))) {
     @helper.form(consolidations.routes.AssociateDucrController.submit(), 'autoComplete -> "off") {
         @helper.CSRF.formField
 
         @components.back_link(consolidations.routes.MucrOptionsController.displayPage())
 
         @components.error_summary(form.errors)
 
-        @components.page_title(Some(messages("associateDucr.title", mucrOptions.mucr)))
-
-        @components.input_text(
-            field = form("ducr"),
-            label = "",
-            hint = Some(messages("associateDucr.reference.hint"))
+        @components.page_title(Some(messages("associate.ucr.title", mucrOptions.mucr)))
+
+        @components.input_radio(
+            field = form("kind"),
+            legend = "",
+            hint = Some(messages("associate.ucr.hint")),
+            inputs = Seq(
+                RadioOption(
+                    keyPrefix = "associate.ucr",
+                    option = AssociateKind.Ducr.formValue,
+                    revealComponent = components.input_text(form("ducr"), "") // FIXME accessability
```

Comment:
Are these getting fixed as part of this task?

Reply:
Done - but probably require little enhancement later